instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/filters/__init__.py

```diff
@@ -0,0 +1,210 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import grpc
+
+
+def _full_method(metadata):
+    name = ""
+    if isinstance(metadata, grpc.HandlerCallDetails):
+        name = metadata.method
+    # NOTE: replace here if there's better way to match cases to handle
+    # grpcext._interceptor._UnaryClientInfo/_StreamClientInfo
+    elif hasattr(metadata, "full_method"):
+        name = metadata.full_method
+    return name
+
+
+def _split_full_method(metadata):
+    name = _full_method(metadata)
+    service, method = os.path.split(name)
+    if service != "":
+        service = os.path.normpath(service)
+        service = service.lstrip("/")
+    return (service, method)
+
+
+def all_of(*args):
+    """Returns a filter function that returns True if all filter functions
+    assigned matches conditions.
+
+    Args:
+        args (function): a list of filter function
+
+    Returns:
+        A filter function that returns True if all filter functions
+    assigned matches conditions.
+    """
+
+    def filter_fn(metadata):
+        for func in args:
+            if not func(metadata):
+                return False
+        return True
+
+    return filter_fn
+
+
+def any_of(*args):
+    """Returns a filter function that returns True if either of filter functions
+    assigned matches conditions.
+
+    Args:
+        args (function): a list of filter function
+
+    Returns:
+        A filter function that returns True if either of filter functions
+    assigned matches conditions.
+    """
+
+    def filter_fn(metadata):
+        for func in args:
+            if func(metadata):
+                return True
+        return False
+
+    return filter_fn
+
+
+def reverse(func):
```

Comment:
I think `not()` or `negate()` would be better names here

Reply:
Thanks for the suggestion! Because `not` is reserved keyword, I take `negate`.